@here/harp-mapview/lib/MapView.ts

```diff
@@ -1610,24 +1682,23 @@ export class MapView extends THREE.EventDispatcher {
      * Returns the zoom level for the given camera setup.
      */
     get zoomLevel(): number {
-        return this.m_zoomLevel;
+        const cameraPitch = MapViewUtils.extractAttitude(this, this.m_camera).pitch;
+        const zoomLevelDistance =
+            this.m_lookAtDistance / Math.cos(Math.min(cameraPitch, Math.PI / 3));
```

Comment:
I think that is wrong. m_lookAtDistance should already be distance to ground divided by cosine of the pitch